Add contributing.md with GH workflow and doc edits/creation details

LZoog opened the pull request:
If we end up moving this repository back into `experimenter` we can create a follow up ticket to make any needed doc updates in this section then.

Closes mozilla/experimenter#4722 / EXP-1027

LZoog commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+Alternatively, consider subscribing to this repository in the [GitHub Slack](https://slack.github.com/) app by searching for it under "Apps" inside Slack.
+
+</div>
+   <img src="../img/github-watch.png" alt="GitHub Watch" style={{ width: '200px', marginLeft: '16px' }} />
```
@jodyheavener I could've added a class or two here instead of `style={{` but I'm not sure that it'd be worth enforcing "no inline styles!!" in the doc hub. It'd be one more thing for reviewers to have to know to lookout for. Whatchu think?

lmorchard replied:
Random notion off the top of my head: Rather than enforce no inline styles, I'd ask if this could have been a class that would come in handy to re-use later. Can also be an optimization to perform later, if a common style comes up often

LZoog replied:
Yeah, it's possible. I'm not sure how often folks will actually be using JSX in MD but I'll go ahead and convert these to classes.

LZoog replied:
OK after converting all these to classes, I like what we've got here much better, so good call. Was probably just me being lazy.

LZoog commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+
+1. Create a new top-level internal link to the doc or modify an existing one
+
+<img src="../img/top-level-link.png" alt="top-level link" style={{ width: '500px' }} />
```
Oooh, @jodyheavener pointed out to me these aren't mobile-friendly, d'oh... I'll fix this up later today.

LZoog commented:
Oh also! I want to modify the front-page README regarding this, and I'll go ahead and add a section for Experimenter engineers to set @jaredlockhart up to add docs related to the new publish status as discussed yesterday.

jodyheavener commented on src/css/custom.css:
```diff
@@ -54,3 +54,7 @@
+
+.flex {
+  display: flex;
+}
```
I wonder how frequently we're going to see CSS changes to accommodate doc layouts 🤔 Is it worth adding a little blurb about how and where to add CSS?

LZoog replied:
I could add a sentence in the doc in this PR about CSS changes, I'll do that with my next push.

jodyheavener commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+
+## Create a New Document
+
+> 👉 [Do this in the GitHub UI](https://github.com/mozilla/experimenter-docs/new/main/docs) 👈
```
More of a nitpick than anything, but these aren't really quotes. I gather you're using `>` to make it stand out, but maybe we could use an admonition or some other button-y type element?

jodyheavener replied:
It would also be good to mention/link to [admonitions](https://v2.docusaurus.io/docs/markdown-features/admonitions) somewhere, maybe near the bit about GFM

LZoog replied:
I tried using an admonition here but it honestly seemed to stand out _too_ much. 😆 I get what you mean about it not being a quote though so maybe I should use one just to be proper, like `tip` where I can add a custom title. Good call out on linking that doc.

LZoog replied:
I used the "note" one and it doesn't look too bad! The "info" one didn't seem quite right and the green one, eh. You can let me know what you think if you don't like it. 😛

<img width="879" alt="image" src="https://user-images.githubusercontent.com/13018240/111006845-27e86d00-8353-11eb-946a-56dcacf02d09.png">

jodyheavener commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+
+<div className="flex">
+<div>
+<b>Notifications</b>
```
Let's try and use as little HTML as possible in Markdown files (this can just be `**Notifications**`)

LZoog replied:
Yep agreed, missed this

jodyheavener commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+
+After committing your changes, you'll be brought back to a pull request view comparing your new branch to `main`. You may need to edit the PR title, but if your files changed look good, click `Create pull request`, and you're done!\*
+
+\*Someone else will need to approve your PR (or give you feedback before approving), and then it will be merged and deployed by GH Pages.
```
I'm not sure it's not in the GFM docs, but Docusaurus supports proper Markdown [footnote syntax](https://www.markdownguide.org/extended-syntax#footnotes) that allows you to link directly to your footnote (and for the footnote to link back to the content). Can we use that instead?

jodyheavener commented on docs/contributing.md:
````diff
@@ -0,0 +1,202 @@
+    ... // other unrelated sidebar items
+```
+
+---
````
Nitpick: the `<hr>` in Dark mode could be a little more subtle if you felt like toning it down in the CSS :)

LZoog replied:
Fixed ;)

jodyheavener commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+4. Create a new external link to a doc in an existing or new category
+
+<div className="flex">
+<img src="../img/category-external-link.png" alt="category link" style={{ width: '300px', marginRight: '32px', marginBottom: '16px' }} />
```
Hrrmm, I kinda have beef with these custom spacing styles. Ideally the docs we write about how to write docs can be easily replicated, and using custom margins on images could be a barrier for someone who wants to make their contribution look like ours, and it also doesn't play well with mobile. Could these styles be added to the CSS as a class like "extra-padding" or "align-right"?

jodyheavener commented on docs/contributing.md:
````diff
@@ -0,0 +1,202 @@
+</div>
+
+```js
+module.exports = {
````
Optional, but could be useful here and good to know for future code snippets: Docusaurus supports snippet titles: https://v2.docusaurus.io/docs/markdown-features/code-blocks#code-title

(I'll let you decide if this is something we should document, I'm not sure how widely used it'd be)

LZoog replied:
_This is cool_

jodyheavener reviewed (commented):
I need to go back through the content, but here are some initial thoughts. This is look really great BTW. I love how you've arranged the images.

k88hudson commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+
+### GitHub UI
+
+The GitHub user interface allows users to make file changes for those that would prefer not to check into the code repository via CLI. See [the GitHub UI Docs](https://docs.github.com/en/github/managing-files-in-a-repository/editing-files-in-your-repository) if you need further details than what's provided here.
```
Maybe, `If you aren't as familiar with git, we recommend using the GitHub user interface to make changes to docs.`

k88hudson commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+
+## Modify an Existing Document
+
+> 👉 [Do this in the GitHub UI](https://github.com/mozilla/experimenter-docs/tree/main/docs) 👈
```
❤️ this quick links

k88hudson commented:
Yeah so this is super cool, love the links to "do this in github" especially. Definitely worth checking if it makes sense to some folks who will be using the github UI.

I think this could use some guidance around like what the process for asking for a review is / how things will get merged

LZoog commented on docs/contributing.md:
```diff
@@ -0,0 +1,202 @@
+
+## GitHub Workflow
+
+All documentation edits and additions will require a pull request into the `main` branch because merged changes will trigger an automatic deployment, but **fear not if you don't prefer checking into code - you probably don't need to.\***
```
I'll rephrase. I mean pulling down the repo locally.

LZoog commented:
Thank you for the comments everyone! I've pushed some updates and believe I've addressed all comments except:

> I think this could use some guidance around like what the process for asking for a review is / how things will get merged

@k88hudson we talked briefly in Slack about this - how do you feel about asking users to come back to their PRs to merge them after someone r+'s? We oftentimes approve things with a comment/suggestion or two that we consider non-blocking, so that seems more ideal in case they want to make a change before merging, but if someone isn't used to hanging out in GitHub, PRs could potentially hang out longer than they need to. As @jodyheavener and I said in Slack, perhaps it's OK to require this at first and see how things go?

Also, should users request the `project-nimbus` team under "Reviewers" (in which case, how would a user request to be added to this team)? Or, do we expect that users that want to receive notifications have configured them for this repo?

LZoog commented on sidebars.js:
```diff
@@ -95,5 +95,6 @@ module.exports = {
       label: "Data Scientists",
       items: ["data-scientists-root"],
     },
+    "contributing",
```
I personally like it at the top-level but at the bottom of the list so it's quite obvious, but if others prefer it to be in a category with "Welcome" I'm fine with that. @k88hudson do you have a preference?

tdsmith commented:
> Also, should users request the project-nimbus team under "Reviewers" (in which case, how would a user request to be added to this team)? Or, do we expect that users that want to receive notifications have configured them for this repo?

Over at mozilla/data-docs the convention is to explicitly tag a reviewer. I'd worry that tagging the team could be noisy.

Requesting a review requires write access, so if they're not on project-nimbus they should do that first. :) "Ask in #nimbus-project" seems good, though.

pdehaan commented:
Re:reviewers, what about using a CODEOWNERS file to auto assign reviewers based on globs? 
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

For example, Tim+Anna are reviewers for any jetstream changes, experimenter core can be reviewed by whomever adds themselves to the codeowners file for whatever paths they volunteer to review/own. I can review PRs that add .svg files.

if we add e2e testing docs, B4hand and I can be notified on changes in those paths.

jodyheavener commented on docs/contributing.md:
```diff
@@ -0,0 +1,215 @@
+          type: "link",
+          label: "My Flying Nimbus",
+          href:
+            "https://magicalyellowcloud.com",,
```
😂

jodyheavener commented on docs/contributing.md:
```diff
@@ -0,0 +1,215 @@
+Alternatively, consider subscribing to this repository in the [GitHub Slack](https://slack.github.com/) app by searching for it under "Apps" inside Slack.
+
+</div>
+   <img src="../img/github-watch.png" alt="GitHub Watch" className="img-xs-right" />
```
Maybe this could be more descriptive? Like "From the repository's 'Notifications settings' menu, select 'All activity' to be notified of all changes to this repository"

jodyheavener reviewed (approved):
This is really great. Content is great. I'm biased but it looks extra great in dark mode. I love the image arrangement.

jodyheavener commented:
Oh while we're here maybe you wanna update or just remove these blurbs about contributing in the main readme? https://github.com/mozilla/experimenter-docs/#using-github

LZoog commented:
@pdehaan I like that idea. Let's create a follow up issue for that since it might warrant some discussion?

I like Tim's suggestion of tagging someone (or the team) and/or "Ask in #nimbus-project" just for now so I can get this merged in. 😎

LZoog commented:
OK, made a few changes including:
* I went ahead and made the "Contributing" link fall under a "Welcome" category as suggested by @jodyheavener, I agree this looks nicer
* I added an "info" blurb around sidebar labels
* I added a "Pull Request Workflow" section
* I edited the front-page documentation
* I added an "Experimenter Engineers" section for Jared to start adding docs to (we can re-order stuff if we want, maybe Engineering should all be grouped together somehow)
* I addressed other comments

Gonna go ahead and pull the trigger here. Anything

pdehaan commented:
> Let's create a follow up issue for that since it might warrant some discussion?

Filed as https://github.com/mozilla/experimenter-docs/issues/11
